src/Servers/Kestrel/test/InMemory.FunctionalTests/Http3/WebTransportTests.cs

```diff
@@ -0,0 +1,129 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Net;
+using System.Net.Http;
+using Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http3;
+using Microsoft.AspNetCore.Testing;
+using Microsoft.Net.Http.Headers;
+using Http3SettingType = Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http3.Http3SettingType;
+
+namespace Microsoft.AspNetCore.Server.Kestrel.Core.Tests;
+
+public class WebTransportTests : Http3TestBase
+{
+    [Fact]
+    public async Task WebTransportHandshake_ClientToServerPasses()
+    {
+        await Http3Api.InitializeConnectionAsync(_noopApplication);
+        var controlStream = await Http3Api.CreateControlStream();
+        var controlStream2 = await Http3Api.GetInboundControlStream();
+
+        var settings = new Http3PeerSettings()
+        {
+            EnableWebTransport = 1,
+            H3Datagram = 1,
+        };
+
+        await controlStream.SendSettingsAsync(settings.GetNonProtocolDefaults());
+        var response1 = await controlStream2.ExpectSettingsAsync();
+
+        await Http3Api.ServerReceivedSettingsReader.ReadAsync().DefaultTimeout();
+
+        Assert.Equal(1, response1[(long)Http3SettingType.EnableWebTransport]);
+
+        var requestStream = await Http3Api.CreateRequestStream();
+        var headersConnectFrame = new[]
+        {
+            new KeyValuePair<string, string>(HeaderNames.Method, "CONNECT"),
+            new KeyValuePair<string, string>(HeaderNames.Protocol, "webtransport"),
```

Comment:
```suggestion
            new KeyValuePair<string, string>(HeaderNames.Protocol, "NOT_webtransport"),
```

This passes, but https://datatracker.ietf.org/doc/html/draft-ietf-webtrans-http3/#section-3.3 says:

> The :protocol pseudo-header field ([RFC8441](https://datatracker.ietf.org/doc/html/rfc8441)) MUST be set to webtransport